react.md

```diff
@@ -1,90 +1,134 @@
+
+# 1. Here is a list of pros and cons to using the React library to build your application -- but some of them are false. Remove the false statements:
+
 - React is a modern, efficient answer to complex UI applications
-- React is a full stack framework for modern web applications
 - React is a flexible library that plays the role of V in an MVC framework
 
- 
- #### 2. What are "smart"(logic) and "dumb"(display) components? Explain the difference and also add why we bother to make the distinction between them.
- 
- 
- //Your Answer
- 
- 
- //Googled Answer
- 
- 
-#### 3. When we use "yarn add ..." in the terminal - what is yarn doing? And what file will always be automatically updated after we add a package with yarn?
- 
- 
- //Your Answer
- 
- 
- //Googled Answer
- 
- 
-#### 5. There are three mistakes in this code that would cause it to break our application. Find the mistakes and fix them:
-
-    import React, { Component } from 'react';
-
-    class Recipes {
-      constructor(props){
+
+
+# 2. What are 'smart'(logic) and 'dumb'(display) components? Explain the difference and also add why we bother to make the distinction between them.
+
+ // Your Answer
+
+ Smart components contain functions and state, essentially controlling the 'brains of the operation'.
+
+ Dumb components only receive information and/or 'call' smart components to action.
+
+ The distinction maintains control of logic flow. Otherwise complex apps with many components would be a cluster F, not knowing who controls what.
+
+ // Googled Answer
+
+ Smart components (container components) keep track of state and care about how the app works.
+
+ Dumb components ('presentational' components) only present something to the DOM. Once that is done, the component is done with it.
+
+ This design pattern is good because it provides separation of concerns. Designers can work on the UI Dumb Component without the need to know all the logic. It also makes the component more reusable, as you can pass different logic to the same component or pass the same logic to different components.
+
+
+
+# 3. When we use 'yarn add ...' in the terminal - what is yarn doing? And what file will always be automatically updated after we add a package with yarn?
+
+ // Your Answer
+
+ Yarn is a program/application installation manager, so 'yarn add' begins this process. So far we've used it for creating React Apps.
+
+ // Googled Answer
+
+ 'Yarn Install' is used to install all dependencies for a project. This is most commonly used when you have just checked out code for a project. This command has been replaced by 'yarn add'.
+
+ This will also update your package.json and your yarn.lock so that other developers working on the project will get the same dependencies as you when they run yarn or yarn install.
+
+
+
+# 4. ???
+
+
+
+# 5. There are three mistakes in this code that would cause it to break our application. Find the mistakes and fix them:
```

Comment:
Great job so far!

Hint: recipes in state is a list. what datatype works for representing a list?